sdk/eventgrid/Azure.Messaging.EventGrid/src/Customization/CloudEvent.cs

```diff
@@ -0,0 +1,79 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using Azure.Core;
+
+namespace Azure.Messaging.EventGrid.Models
+{
+    /// <summary> Properties of an event published to an Event Grid topic using the CloudEvent 1.0 Schema. </summary>
+    public partial class CloudEvent
+    {
+        /// <summary> Initializes a new instance of CloudEvent. </summary>
+        /// <param name="source"> Identifies the context in which an event happened. The combination of id and source must be unique for each distinct event. </param>
+        /// <param name="type"> Type of event related to the originating occurrence. </param>
+        public CloudEvent(string source, string type)
+        {
+            if (source == null)
```

Comment:
I think Soren mentioned in the Teams chat, that source isn't actually required by the service even though it is required in the spec. Does the service set it for us if left blank?